src/core/src/core_logic/PatchAssessor.py

```diff
@@ -81,15 +88,26 @@ def start_assessment(self):
                 else:
                     error_msg = 'Error retrieving available patches: ' + repr(error)
                     self.composite_logger.log_error(error_msg)
+                    self.write_assessment_perf_logs(number_of_tries, Constants.TaskStatus.FAILED, error_msg)
                     self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.DEFAULT_ERROR)
                     if Constants.ERROR_ADDED_TO_STATUS not in repr(error):
                         error.args = (error.args, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
                     self.status_handler.set_assessment_substatus_json(status=Constants.STATUS_ERROR)
                     raise
 
+        self.write_assessment_perf_logs(number_of_tries, Constants.TaskStatus.SUCCEEDED, "")
         self.composite_logger.log("\nPatch assessment completed.\n")
         return True
 
+    def write_assessment_perf_logs(self, number_of_tries, task_status, error_msg):
+        try:
+            assessment_perf_log = {Constants.PerfLogTrackerParams.TASK: Constants.ASSESSMENT, Constants.PerfLogTrackerParams.TASK_STATUS: str(task_status),
+                                   Constants.PerfLogTrackerParams.ERROR_MSG: error_msg, Constants.PerfLogTrackerParams.PACKAGE_MANAGER: self.package_manager_name,
+                                   Constants.PerfLogTrackerParams.NUMBER_OF_TRIALS: str(number_of_tries)}
+            self.stopwatch.stop_and_write_telemetry(str(assessment_perf_log))
+        except Exception as error:
+            self.composite_logger.log_debug("Error in writing assessment performance logs. Error is: " + repr(error))
```

Comment:
Everything else looks good, one minor comment

We aren't throwing Exceptions from Stopwatch class anymore, so this scenario should not occur. The only place throwing an Exception is get_percentage_maintenance_window_used() which is only called within write_installer_perf_logs()

Reply:
Removed the catch exception and removed corresponding unit test